### ISSUE TITLE
Use wsl.exe to launch WSL shell instead of bash.exe

### ISSUE THREAD

**Issue by gtritchie** [rstudio/rstudio#6855]
On Windows-10 builds starting with Spring 2018 update, the RStudio Terminal should detect and launch WSL via `wsl.exe` instead of `bash.exe`. Use of `bash.exe`  (which isn't actually bash but the WSL launcher) is considered deprecated in these newer versions of WIndows.

Deprecation of bash mentioned here: https://docs.microsoft.com/en-us/windows/wsl/interop

**Comment by gtritchie** [comment]
Bump

**Comment by Danny-dK** [comment]
Is this still being worked on? I get:

[bsh_rstudio.webm](https://github.com/rstudio/rstudio/assets/45395070/b7fb6a18-c97b-477c-bb6b-c11f238bfc4f)

Opening a terminal that is set to 'Bash (Windows Subsystem for Linux)' seems to not do anything. For less then a second I can see it is trying to load, but then I can't do anything in this screen. If I set the terminal in global options to custom shell binary and point to C:/Windows/System32/wsl.exe everything works fine.

**Comment by Danny-dK** [comment]
@kevinushey The thing is that if I go to `System32` and start bash.exe, that still works. A shell window is opening that starts my Ubuntu 20.04LTS distro under WSL2. Just not starting in Rstudio terminal.

**Comment by Danny-dK** [comment]
Hmm, if I set the custom terminal location to wsl.exe everything works, if I set it to bash.exe it does not work. Perhaps point rstudio to wsl instead of bash.exe when `win-wsl-bash` is used as the option? (I don't know if it is these lines https://github.com/rstudio/rstudio/blob/8244ebc49cc3188f04aa26bb2c2377aeda315817/src/cpp/session/modules/SessionTerminalShell.cpp#L111C3-L123C55)

**Comment by gtritchie** [comment]
> ...Perhaps point rstudio to wsl instead of bash.exe when `win-wsl-bash` is used as the option?

Yes, that is what this issue was opened for. I'll look at finally making that change (better late than never?).

**Comment by Danny-dK** [comment]
> Yes, that is what this issue was opened for.

Ah sorry, then I misunderstood. Hope that the issue since 2020 will be resolved soon.

**Comment by gtritchie** [comment]
> Ah sorry, then I misunderstood. Hope that the issue since 2020 will be resolved soon.

No worries, sorry this issue hasn't been fixed already! I'll be working on it soon.

**Comment by gtritchie** [comment]
It turns out the problem isn't wsl.exe versus bash.exe! This was actually a regression and changing the code to use wsl.exe isn't sufficient to fix it (fails the same way).

I've opened https://github.com/rstudio/rstudio/issues/13918 to track fixing that; this issue goes back to just being for tracking switching from wsl.exe to bash.exe (which we should still do, theoretically).
